lib/streamaggr/streamaggr.go

```diff
@@ -529,8 +529,8 @@ func newAggregator(cfg *Config, pushFunc PushFunc, ms *metrics.Set, opts Options
 		return nil, fmt.Errorf("`outputs` list must contain at least a single entry from the list %s; "+
 			"see https://docs.victoriametrics.com/stream-aggregation/", supportedOutputs)
 	}
-	aggrStates := make([]aggrState, len(cfg.Outputs))
-	for i, output := range cfg.Outputs {
+	aggrStates := make(map[string]aggrState, len(cfg.Outputs))
+	for _, output := range cfg.Outputs {
```

Comment:
This could be a breaking change. As it would make the following config incorrect:
```
- match:request_duration_seconds
  interval: 30s
  outputs:  ["quantiles(0.50)", "quantiles(0.99)"]
```
For this configuration, vmagent will produce only one quantile and ignore another. When before both quantiles will be calculated.

Should we validate duplicates and repeating `quantiles` items in `outputs`?

Reply:
Good catch!
added validation and mentioned in docs.